.github/workflows/ci.yaml

```diff
@@ -22,11 +22,12 @@ concurrency:
 
 jobs:
   test:
-    name: "Unit Tests and Type Checking"
+    name: "${{ matrix.os }} :: ${{ matrix.version }} :: Unit Tests and Type Checking"
```

Comment:
Argh GitHub. This was also being used for a required check matching `Unit Tests and Type Checking (3.11, ubuntu-22.04)`, but that check is actually picking up `TK CI / Unit Tests and Type Checking (3.11, ubuntu-22.04)` also.

I might just copy this snippet from IREE to have a "ci-summary" job with a unique name we can set as required: https://github.com/iree-org/iree/blob/7177c29f9b2d9e255b63987f5dfff174ec2afc2f/.github/workflows/ci.yml#L201-L236